include/drivers/arm/tzc380.h

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2018, ARM Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef __TZC380_H__
+#define __TZC380_H__
+
+#include <tzc_common.h>
+
+#define BUILD_CONFIG_OFF	0x000
+#define ACTION_OFF		0x004
+#define LOCKDOWN_RANGE_OFF	0x008
+#define LOCKDOWN_SELECT_OFF	0x00C
+#define INT_STATUS		0x010
+#define INT_CLEAR		0x014
+
+#define FAIL_ADDRESS_LOW_OFF	0x020
+#define FAIL_ADDRESS_HIGH_OFF	0x024
+#define FAIL_CONTROL_OFF	0x028
+#define FAIL_ID			0x02c
+
+#define SPECULATION_CTRL_OFF	0x030
+#define SECURITY_INV_EN_OFF	0x034
+
+#define REGION_SETUP_LOW_OFF(n)	(0x100 + n * 0x10)
+#define REGION_SETUP_HIGH_OFF(n) (0x104 + n * 0x10)
+#define REGION_ATTRIBUTES_OFF(n) (0x108 + n * 0x10)
+
+#define BUILD_CONFIG_AW_SHIFT	8
+#define BUILD_CONFIG_AW_MASK	0x3f
+#define BUILD_CONFIG_NR_SHIFT	0
+#define BUILD_CONFIG_NR_MASK	0xf
+
+#define ACTION_RV_SHIFT		0
+#define ACTION_RV_MASK		0x3
+#define  ACTION_RV_LOWOK	0x0
+#define  ACTION_RV_LOWERR	0x1
+#define  ACTION_RV_HIGHOK	0x2
+#define  ACTION_RV_HIGHERR	0x3
+
+#ifndef BIT
+#define BIT(nr)			(1 << (nr))
+#endif
+
+/* Speculation is enabled by default. */
+#define SPECULATION_CTRL_WRITE_DISABLE	BIT(1)
```

Comment:
The destination register is a 32-bit register, so this should use ``BIT_32()`` instead.